firefox-bootstrap/bootstrap.js

```diff
@@ -0,0 +1,259 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+let { classes: Cc, interfaces: Ci, utils: Cu } = Components;
+
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "CustomizableUI",
+  "resource:///modules/CustomizableUI.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "Services",
+  "resource://gre/modules/Services.jsm");
+
+const PREF_WC_REPORTER_ENABLED = "extensions.webcompat-reporter.enabled";
+const WEBCOMPAT_DOMAIN = "webcompat.com";
+const WIDGET_ID = "webcompat-reporter-button";
+
+class TabListener {
+  constructor(browser) {
+    this.browser = browser;
+    this.addListeners();
+  }
+
+  get QueryInterface() {
+    return XPCOMUtils.generateQI(["nsIWebProgressListener",
+      "nsISupportsWeakReference"]);
+  }
+
+  addListeners() {
+    this.browser.addProgressListener(this);
+    this.browser.tabContainer.addEventListener("TabSelect",
+      this.setButtonState);
+  }
+
+  removeListeners() {
+    this.browser.removeProgressListener(this);
+    this.browser.tabContainer.removeEventListener("TabSelect",
+      this.setButtonState);
+  }
+
+  onStateChange(webProgress, request, flag, status) {
+    if (flag & Ci.nsIWebProgressListener.STATE_STOP) {
+      // about:new-tab loads don't have an originalURI property, but
+      // have a request.name of about:document-onload-blocker
+      let scheme = request.originalURI ? request.originalURI.scheme
+                 : request.name;
+      this.setButtonState(scheme);
+    }
+  }
+
+  setButtonState(arg) {
+    // Did we get here from a TabSelect event (arg = event object)
+    // or manually via onStateChange (arg = string)?
+    let scheme = typeof arg === "string" ? arg :
+                 arg.target.linkedBrowser.currentURI.scheme;
+    let button = CustomizableUI.getWidget(WIDGET_ID);
+    let isInvalidScheme = ["about", "chrome", "file", "resource", "view-source"].
+      some((prefix) => scheme.startsWith(prefix));
+
+    button.disabled = isInvalidScheme;
+  }
+}
+
+let windowListener = new class {
+  onOpenWindow(window) {
+    let win = window.QueryInterface(Ci.nsIInterfaceRequestor). // eslint-disable-line new-cap
```

Comment:
Cool, let's do it. I'm gonna remove the `no-unused-vars` comments as well before m-c review (NOTE TO SELF).